src/App.tsx

```diff
@@ -1,23 +1,83 @@
-import * as React from 'react';
-import { ChakraProvider, Box, Text, Link, VStack, Code, Grid, theme } from '@chakra-ui/react';
-import { ColorModeSwitcher } from './ColorModeSwitcher';
-import { Logo } from './Logo';
+import React, {useState} from 'react';
+import { ChakraProvider, Text, Box, Button, Center /* Text, Link, VStack, Code, Grid, theme*/ } from '@chakra-ui/react';
+import theme from './theme'
+import Header from './Components/Header'
+import Board from './Components/Board'
+function returnInitialState(storageKey: string) {
+  try {
+    // Get from local storage by key
+    const item = window.localStorage.getItem(storageKey);
+    // Parse stored json or if none return an empty object
+    return item ? JSON.parse(item) : 0;
+  } catch (error) {
+    // If error also return an empty object
+    console.log(error);
+    return 0;
+  }
+}
 
-export const App = () => (
-  <ChakraProvider theme={theme}>
-    <Box textAlign="center" fontSize="xl">
-      <Grid minH="100vh" p={3}>
-        <ColorModeSwitcher justifySelf="flex-end" />
-        <VStack spacing={8}>
-          <Logo h="40vmin" pointerEvents="none" />
-          <Text>
-            Edit <Code fontSize="xl">src/App.tsx</Code> and save to reload.
-          </Text>
-          <Link color="teal.500" href="https://chakra-ui.com" fontSize="2xl" target="_blank" rel="noopener noreferrer">
-            Learn Chakra
-          </Link>
-        </VStack>
-      </Grid>
+function useLocalStorage(storageKey: string) {
+  const [storedValue, setStoredValue] = useState(
+    returnInitialState(storageKey)
+  );
+  
+  const setValue = (value: any) => {
+    try {
+    // Allow value to be a function so we have same API as useState
+      const valueToStore =
+      value instanceof Function ? value(storedValue) : value;
+      // Save to local storage
+      window.localStorage.setItem(storageKey, JSON.stringify(valueToStore));
+      // Save state
+      setStoredValue(valueToStore);
+    } catch (error) {
+      // A more advanced implementation would handle the error case
+      console.log(error);
+    }
+  };
+  
+  return [storedValue, setValue]
+}
+export function App() {
+
+  const [player, setPlayer] = useState<string>("X")
+  const [p1Score, setP1Score] = useLocalStorage("setP1Score")
+  const [p2Score, setP2Score] = useLocalStorage("setP2Score")
+  const [won, setWon] = useState<boolean>(false);
+  const [whoWon, setWhoWon] = useState<string>("Player 1 Won!")
+  const [headColor, setHeadColor] = useState<string>('white');
```

Comment:
Does this need to be stateful value? Can you not compute the current head color based off other stateful values?